core/src/Platforms/iOS/iOSPlatformProxy.cs

```diff
@@ -97,13 +97,17 @@ public void ValidateRedirectUri(Uri redirectUri, RequestContext requestContext)
             {
                 if (Constants.DefaultRedirectUri.Equals(redirectUri.AbsoluteUri, StringComparison.OrdinalIgnoreCase))
                 {
-                    // TODO: Need to use CoreExceptionFactory here...?
-                    //throw new MsalException(MsalError.RedirectUriValidationFailed, "Default redirect URI - " + Constants.DefaultRedirectUri +
-                    //                                                               " cannot be used on iOS platform");
-                    throw new InvalidOperationException(
-                        $"Default redirect URI - {Constants.DefaultRedirectUri} cannot be used on iOS platform");
+                    throw CoreExceptionFactory.Instance.GetClientException(
```

Comment:
@jmprieur - should we add more defaults? 

Currently, the logic across platforms is as follows: 

- on UWP, ADAL will use "http://sso" if the redirect uri is NULL
- on iOS and Android, MSAL will throw if you are trying to pass NULL or "urn:obo:..." standard uri
- everywhere else, throw if redirect uri is NULL

Reply:
@bgavrilMS :
- I was not aware of http://sso. Would you know if it has a special signification for WAB?  In any case we cannot change it in ADAL.NET as this would be a breaking change. In MSAL.NET I thought this was "urn:obo:..." ?
- on iOS and Android, for MSAL.NET, I'm pretty sure that the default is msal{clientId}://auth?
- for pure client credential flows we don't need a redirect URI (but we do for all the other confidential client flows). Is there something we can do for not checking the redirect URI in client credentials flows (with app password and certificate)?

Reply:
Thank you @jmprieur, I made more changes to reflect this. Indeed, the defaults are exactly as you say. 
On UAP, ADAL uses WEB so the default uri is "http://sso"